packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -20,6 +20,7 @@ type ResolverOptions = {
   moduleDirectory?: Array<string>;
   paths?: Array<Config.Path>;
   rootDir?: Config.Path;
+  packageFilter?: (pkg: any, pkgfile: string) => any;
```

Comment:
```suggestion
  packageFilter?: (pkg: unknown, pkgfile: string) => any;
```

or pull the type from `resolve` itself